test/functional/tests/stress/test_stress_without_io.py

```diff
@@ -0,0 +1,331 @@
+        description: Validate the ability of CAS to reload modules in the loop.
+        pass_criteria:
+          - No system crash while reloading CAS modules.
+          - CAS modules reloads with no errors.
+          - No data corruption.
+    """
+    with TestRun.LOGGER.step("Prepare cache and core. Create test file and count it's checksum."):
+        cache, core, md5_before_load, size_before_load, permissions_before_load = \
+            prepare_with_file_creation(cache_mode)
+
+    with TestRun.LOGGER.step("Save current cache configuration."):
+        cache_config = cache.get_cache_config()
+
+    TestRun.LOGGER.start_group(
+        f"Reload CAS modules and check loaded cache configuration {iterations_per_config} times.")
+    for i in range(0, iterations_per_config):
+        TestRun.LOGGER.start_iteration(f"Iteration {i + 1}/{iterations_per_config}")
+        with TestRun.LOGGER.step("Stop cache."):
+            cache.stop()
+            caches_count = len(casadm_parser.get_caches())
+            if caches_count != 0:
+                TestRun.fail(f"Expected caches count: 0; Actual caches count: {caches_count}.")
+            cores_count = len(casadm_parser.get_cores(cache.cache_id))
+            if cores_count != 0:
+                TestRun.fail(f"Expected cores count: 0; Actual cores count: {cores_count}.")
+        with TestRun.LOGGER.step("Reload CAS modules."):
+            cas_module.reload_all_cas_modules()
+        with TestRun.LOGGER.step("Load cache."):
+            cache = casadm.load_cache(cache.cache_device)
+            caches_count = len(casadm_parser.get_caches())
+            if caches_count != 1:
+                TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+            cores_count = len(casadm_parser.get_cores(cache.cache_id))
+            if cores_count != 1:
+                TestRun.fail(f"Expected cores count: 1; Actual cores count: {cores_count}.")
+        with TestRun.LOGGER.step("Validate cache configuration."):
+            if cache.get_cache_config() != cache_config:
+                TestRun.fail("Cache configuration is different than before reloading modules.")
+        TestRun.LOGGER.end_iteration()
+    TestRun.LOGGER.end_group()
+
+    with TestRun.LOGGER.step("Check md5 of test file."):
+        check_files(core, size_before_load, permissions_before_load, md5_before_load)
+    casadm.stop_all_caches()
+
+
+def check_files(core, size_before, permissions_before, md5_before):
+    TestRun.LOGGER.info("Checking file md5.")
+    core.mount(mount_point)
+    file_after = fs_utils.parse_ls_output(fs_utils.ls(test_file_path))[0]
+    md5_after = file_after.md5sum()
+    if md5_before != md5_after:
+        TestRun.LOGGER.error(f"Md5 before ({md5_before}) and after ({md5_after}) are different.")
+
+    if permissions_before.user != file_after.permissions.user:
+        TestRun.LOGGER.error(f"User permissions before ({permissions_before.user}) "
```

Comment:
Did you mean `TestRun.fail()`? `TestRun.LOGGER.error()` doesn't cause test to fail, so it will be marked as pass despite errors were detected. I guess this is not desired behavior.

Reply:
I meant LOGGER.error(), because I don't want to interrupt test execution. Error message in logger causes overall test result to be failed at the end of the test, but doesn't interrupt the execution and that's the expected behavior. (especially in this method, I want to check all 
parameters and have results of those checks in result log file. If I'll use TestRun.fail() here, test execution will be finished and there will be no information in log, if other checks passed or not.